etc/docker/Dockerfile

```diff
@@ -0,0 +1,74 @@
+FROM fpco/stack-build:lts-16.12 as deps-tamarin
+RUN mkdir /opt/build
+WORKDIR /opt/build
+ENV DEBIAN_FRONTEND=noninteractive
+
+COPY stack.yaml stack.yaml.lock tamarin-prover.cabal /opt/build/
+RUN mkdir lib lib/export lib/theory lib/sapic lib/tools lib/term
+COPY lib/sapic/tamarin-prover-sapic.cabal lib/sapic/
+COPY lib/term/tamarin-prover-term.cabal lib/term/
+COPY lib/theory/tamarin-prover-theory.cabal lib/theory/
+COPY lib/utils/tamarin-prover-utils.cabal lib/utils/
+RUN stack build --system-ghc --dependencies-only --ghc-options="+RTS -M600M" -j1
+
+FROM fpco/stack-build:lts-16.12 as build-tamarin
+# Copy compiled dependencies from previous stage
+COPY --from=deps-tamarin /root/.stack /root/.stack
```

Comment:
Since this is using `stack` anyway, wouldn't it be preferable to use `--stack-root` (or `$STACK_ROOT`) to move this to a defined location outside `/root`.